test/suites/wdio.conf.js

```diff
@@ -95,75 +96,25 @@ exports.config = {
 	// define all tests
 	specs: [ './specs/**/*.js' ],
 
-	suites: {
-
-		// example-specs
-		example: [
-			'./specs/quickstatements/*.js',
-			'./specs/repo/queryservice.js',
-			'./specs/elasticsearch/*.js'
-		],
-
-		// bundle-specs
-		repo: [
-			'./specs/repo/*.js',
-			'./specs/repo/extensions/*.js'
-		],
-		repo_client: [
-			'./specs/repo_client/*.js',
-			'./specs/repo_client/extensions/*.js'
-		],
-		fedprops: [
-			'./specs/fedprops/*.js'
-		],
-		pingback: [
-			'./specs/pingback/*.js'
-		],
-		quickstatements: [
-			'./specs/repo_client/interwiki-links.js',
-			'./specs/quickstatements/*.js'
-		],
-		elasticsearch: [
-			'./specs/elasticsearch/*.js'
-		],
-		confirm_edit: [
-			'./specs/confirm_edit/*.js'
-		],
-
-		// base-specs
-		base__repo: [
-			'./specs/repo/api.js',
-			'./specs/repo/property.js',
-			'./specs/repo/special-item.js',
-			'./specs/repo/special-property.js',
-			'./specs/repo/queryservice.js'
-		],
-		base__repo_client: [
-			'./specs/repo_client/interwiki-links.js',
-			'./specs/repo_client/item.js',
-			'./specs/repo/api.js'
-		],
-		base__fedprops: [
-			'./specs/fedprops/*.js'
-		],
-		base__pingback: [
-			'./specs/pingback/*.js'
-		],
-
-		pre_upgrade: [
-			'./specs/repo/api.js',
-			'./specs/upgrade/pre-upgrade.js',
-			'./specs/upgrade/queryservice-pre-and-post-upgrade.js'
-		],
-
-		post_upgrade: [
-			'./specs/repo/api.js',
-			'./specs/upgrade/post-upgrade.js',
-			'./specs/upgrade/queryservice-pre-and-post-upgrade.js',
-			'./specs/upgrade/queryservice-post-upgrade.js'
-		]
-
-	},
+	suites: Object.fromEntries(
+		fs
+			.readdirSync( __dirname )
```

Comment:
Something to think about here is that for right now, and I am not sure that will stay true always, this `wdio.conf.js` gets loaded anew each run for each suite. So it might be more explicit/clear to only load the `${resultDir}/${suite}.conf.js` for now and add this traversing once or if this ever changes. * I hadn't thought about this detail before, and I realize I lead you down the path of "composing from all the directories". I would in the end prefer though to only load this single file so there is less cognitive dissonance with `resultsDir` which explicitly ties the conf for now to a particular suite run... if that makes sense.

Reply:
Updated!